pkg/oidc/oidc.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oidc
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/hcl/v2"
+)
+
+const (
+	defaultClientID      = "gangway"
+	defaultUsernameClaim = "email"
+	defaultGroupsClaim   = "groups"
+)
+
+// Config deals with providing OIDC related fields to the Lokomotive
+// cluster for OIDC based Authentication via Dex and Gangway.
+type Config struct {
+	IssuerURL     string `hcl:"issuer_url,optional"`
```

Comment:
This shouldn't be `optional`, right?

Reply:
Or is it empty because you set it later here https://github.com/kinvolk/lokomotive/pull/182/commits/fa6a8c66f6fd34300a098f8ee8ab5b321197ec60#diff-85ba19a02a388be22d6e76efc96c06eaR87-R89 ?

Reply:
It is optional because we set it to `https://dex.<cluster-name>.<dns-zone>` if the user hasn't provided any input.